Library/Homebrew/test/testing_env.rb

```diff
@@ -20,6 +20,25 @@
   abort "Run `bundle install` or install the mocha and minitest gems before running the tests"
 end
 
+if ENV["HOMEBREW_TESTS_COVERAGE"]
+  # Ensure all Ruby files are `require`d for an accurate coverage measurement
+  Dir[File.expand_path("../../**/*.rb", __FILE__)].each do |f|
+    excluded_core_files = [
+      # scripts
+      "build.rb", "postinstall.rb", "test.rb",
+      # mocked files
+      "config.rb", "global.rb",
+    ]
+
+    next if excluded_core_files.any? { |file| f.end_with?("Homebrew/#{file}") }
+
+    # ignore test files
+    next if f.include?("/test/")
+
+    require f
+  end
+end
```

Comment:
By hack, I mean I rarely see other project to require all files using file glob. (On the hand, manually requiring all the code using `require "libname"` is quite common. )

My concerns are (a) this can raise hard to debug problem, as the order matters. (b) brew tests might give false postive result, because we don't require all codes like this outside test.